unstructured/partition/doc.py

```diff
@@ -63,52 +63,74 @@ def partition_doc(
         This information will be reflected in elements' metadata and can be be especially
         useful when partitioning a document that is part of a larger document.
     """
-    # Verify that only one of the arguments was provided
-    if filename is None:
-        filename = ""
```

Comment:
There is no need for a `None` filename value to be converted to `str`. Ditch this.